sql/teradata/TeradataSQLDDLParser.g4

```diff
@@ -0,0 +1,2079 @@
+                                          )
+                  ;
+
+alter_range_expr
+    : (RANGE|RANGE_L) ( BETWEEN range_expr_3 (',' range_expr_3)*
+                        ( ',' NO RANGE ((OR|',') UNKNOWN)?
+                        | ',' UNKNOWN
+                        )?
+                      | NO RANGE ((OR|',') UNKNOWN)?
+                      | UNKNOWN
+                      | WHERE logical_expr
+                      )
+    ;
+
+with_isolated_loading_alter_form : with_isolated_loading (FOR (ALL|INSERT|NONE))? (USING FAST MODE (ON|OFF))? ;
+
+join_index_add_option
+    : ADD ( (COLUMN|ROW|SYSTEM) '(' name=unqualified_name ')' (NO? AUTO COMPRESS)?
+          | '(' name=unqualified_name ')' NO? AUTO COMPRESS
+          )
+    ;
+
+/***********************
+    ALTER TYPE statement
+*/
+alter_type_stat
+    : ALTER TYPE sysudtlib? udt_name
+        ( add_attribute_clause
+        | add_method_clause
+        | add_specific_method_clause
+        | drop_attribute_clause
+        | drop_method_clause
+        | drop_specific_method_clause
+        | COMPILE ONLY?
+        )
+    ;
+
+add_attribute_clause : ADD ATTRIBUTE type_attribute_spec (',' type_attribute_spec)*;
+
+add_method_clause : ADD (INSTANCE|CONSTRUCTOR)? METHOD add_method_spec (',' add_method_spec)* ;
+
+add_specific_method_clause : ADD SPECIFIC METHOD add_specific_method_spec (',' add_specific_method_spec)* ;
+
+drop_attribute_clause : DROP ATTRIBUTE attribute_name (',' attribute_name)* ;
+
+drop_method_clause : DROP (INSTANCE|CONSTRUCTOR)? METHOD sysudtlib? method_name (',' sysudtlib? method_name)* ;
+
+drop_specific_method_clause : DROP SPECIFIC METHOD sysudtlib? method_name (',' sysudtlib? method_name)* ;
+
+add_method_spec
+    : sysudtlib? method_name '(' ( data_type (',' data_type)* )? ')'
+              RETURNS returns_parameter_spec
+              (CAST FROM cast_from_data_type=data_type )?
+              (SPECIFIC sysudtlib? specific_method_name=method_name)?
+              (method_attr* method_language_spec method_attr+|method_attr+ method_language_spec method_attr*)
+    ;
+
+add_specific_method_spec
+    : sysudtlib? specific_method_name=method_name ( '(' data_type (',' data_type)* ')' )? FOR sysudtlib? udt_name
+    ;
+
+/***********************************
+    BEGIN ISOLATED LOADING statement
+*/
+begin_isolated_loading_stat
+    : BEGIN CONCURRENT? ISOLATED LOADING ON table_name (',' table_name)*
+      USING QUERY_BAND query_band=char_string_literal // 'LDILoadGroup=value;'
+      (IN? (SINGLE|MULTIPLE) SESSION)?
+    ;
+
+/**************************
+    BEGIN LOGGING statement
+*/
+begin_logging_stat
+    : BEGIN LOGGING DENIALS? (WITH TEXT)?
+      ON logging_frequency
+      (FOR CONSTRAINT constraint_name=object_name)?
+      (ALL|operation (',' operation)*)
+      (FOR CONSTRAINT constraint_name=object_name)?
+      (BY user_name (',' user_name)* )?
+      (ON logging_item (',' logging_item)* )?
+    ;
+
+operation
```

Comment:
Please sort and uniq this list. You have duplicates, which is causing ambiguity to be detected in examples/ddl/logging/begin_logging.sql. For example, two `INSERT`, two `SELECT`, two `UPDATE`.